api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -45,6 +45,22 @@ message BootstrapConfigDump {
   google.protobuf.Timestamp last_updated = 2;
 }
 
+// This is a general purpose proto for dynamic configuration to track information about both
+// currently loaded configuration and the most recent attempt to load configuration.
+message ConfigUpdateState {
+  // The name of the entity whose state is tracked here.
+  string name = 1;
```

Comment:
So as commented below I don't think that's the right choice here

  // Review: Should this be per DynamicListener?
  // Really per name we want the latest where warming > active > draining and
  // recent loads get mixed in so I think global is correct.

It may be that it makes sense for the other DSs and the listener is weird because it has three versions live, but I think it makes sense to have one state not 4x, so I think it ought to be global 
Basically if we want to know current status you'd have to figure out which of active, warming and draining to look at, and none of those would include "failed to load initially" where I think we want that staus visible, so I think global is the right thing here?

Reply:
Arguably, from a top-down perspective, we should just have one way of getting at data that belongs to any listener. Can we just have a listener list full stop, and an enum indicating whether warming/active/draining? This is v2alpha, so we can just make the change without "breaking" the API. @mattklein123 thoughts?

Reply:
Stepping back, what are we trying to achieve here? Basically we just want to know for a named listener what its state is? If so yes I think it's reasonable to rejigger the output so we have a list of listeners, within each listener we have its state, and then if there is a warming/draining secondary config would could optionally print that?

Reply:
I want to be able to at a glance for any reloadable component in the system see
1) details of the last successful load (when it happened, time it took, config which we tried to load) and 
2) details iff the last load failed (when it happened, time it took to fail, [eventually] config we tried to load, error message)

So if we have [ successful load ] [ failed load] I'd want to see both, but if we have [failed load] [ successful load] we'd have a green dashboard

I don't mind doing a larger refactor where we track listeners, each listener has a LoadState (which may look like a combination of ConfigUpdateState and DynamicListener) for last_successful_load and last_failed_load, but it still begs the question if a listener has never loaded, how we want to handle that, because I'd still like to have "recent" failures visible on the dashboard even if a listener never successfully loaded.  Harvey, would you suggest the listener list just have {warming / active / draining / inoperational } where again we'd have to reap inoperational by [some configuration values] so we didn't just have a slow leak with stale failures?